presto-hive/src/test/java/com/facebook/presto/hive/hudi/HudiTestingDataGenerator.java

```diff
@@ -0,0 +1,233 @@
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+import static java.util.Objects.requireNonNull;
+import static org.apache.hudi.common.model.HoodieTableType.COPY_ON_WRITE;
+import static org.apache.hudi.common.model.HoodieTableType.MERGE_ON_READ;
+
+public class HudiTestingDataGenerator
+{
+    private static final String OWNER_PUBLIC = "public";
+    private static final MetastoreContext METASTORE_CONTEXT = new MetastoreContext("test_user", "test_queryId", Optional.empty(), Optional.empty(), Optional.empty(), false, HiveColumnConverterProvider.DEFAULT_COLUMN_CONVERTER_PROVIDER);
+    private static final PrincipalPrivileges PRINCIPAL_PRIVILEGES = new PrincipalPrivileges(ImmutableMultimap.of(), ImmutableMultimap.of());
+
+    private static final StorageFormat STORAGE_FORMAT_COPY_ON_WRITE = StorageFormat.create(
+            ParquetHiveSerDe.class.getName(),
+            HoodieParquetInputFormat.class.getName(),
+            MapredParquetOutputFormat.class.getName());
+    private static final StorageFormat STORAGE_FORMAT_MERGE_ON_READ = StorageFormat.create(
+            ParquetHiveSerDe.class.getName(),
+            HoodieParquetRealtimeInputFormat.class.getName(),
+            MapredParquetOutputFormat.class.getName());
+
+    public static final List<Column> DATA_COLUMNS = ImmutableList.of(
+            column("volume", HIVE_LONG),
+            column("ts", HIVE_STRING),
+            column("symbol", HIVE_STRING),
+            column("year", HIVE_INT),
+            column("month", HIVE_STRING),
+            column("high", HIVE_DOUBLE),
+            column("low", HIVE_DOUBLE),
+            column("key", HIVE_STRING),
+            column("date", HIVE_STRING),
+            column("close", HIVE_DOUBLE),
+            column("open", HIVE_DOUBLE),
+            column("day", HIVE_STRING));
+    public static final List<Column> PARTITION_COLUMNS = ImmutableList.of(column("dt", HIVE_STRING));
+    public static final List<Column> HUDI_META_COLUMNS = ImmutableList.of(
+            column("_hoodie_commit_time", HiveType.HIVE_STRING),
+            column("_hoodie_commit_seqno", HiveType.HIVE_STRING),
+            column("_hoodie_record_key", HiveType.HIVE_STRING),
+            column("_hoodie_partition_path", HiveType.HIVE_STRING),
+            column("_hoodie_file_name", HiveType.HIVE_STRING));
+
+    private final Path dataDir;
+    private final ExtendedHiveMetastore metastore;
+    private final String schemaName;
+
+    public HudiTestingDataGenerator(ExtendedHiveMetastore metastore, String schemaName, Path dataDir)
+    {
+        this.metastore = requireNonNull(metastore, "metastore is null");
+        this.schemaName = requireNonNull(schemaName, "catalogName is null");
+        this.dataDir = requireNonNull(dataDir, "dataDir is null");
+    }
+
+    public void generateData()
+    {
+        try {
+            InputStream stream = Resources.getResource("hudi-testing-data.zip").openStream();
```

Comment:
Should we close this stream in a finally block, or maybe try-with-resources?

Reply:
Good catch, thank you!